src/components/Toast/ToasterContext.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  ReactNode,
+  createContext,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react"
+import { createPortal } from "react-dom"
+import { v4 } from "uuid"
+import { Transition, TransitionGroup } from "react-transition-group"
+import { Toast as ToastType } from "../../types/Toast"
+import { useKeyPress } from "../../hooks/useKeyPress"
+import {
+  ToasterConfig,
+  getToastPositionStyles,
+  getToastTransitionStyles,
+} from "./utils"
+import { Toast } from "./Toast"
+
+interface ToasterProviderProps {
+  children: ReactNode
+  config: ToasterConfig
+}
+
+interface ContextProps {
+  addToast: (toast: Omit<ToastType, "id">) => void
+  removeAll: () => void
+}
+
+export const ToasterContext = createContext<ContextProps>({
+  addToast: () => null,
+  removeAll: () => null,
+})
+
+export function ToasterProvider({ children, config }: ToasterProviderProps) {
+  const { animationDuration, duration, behavior, position } = config.getConfig()
+  const [toasts, setToasts] = useState<ToastType[]>([])
+  const timer = useRef<NodeJS.Timeout>()
+
+  const removeToast = useCallback((id: string) => {
+    setToasts((toasts) => toasts.filter((toast) => toast.id !== id))
+  }, [])
+
+  const removeOldest = useCallback(() => {
+    if (toasts.length === 0) {
+      return
+    }
+    setToasts((toasts) => toasts.slice(1))
+  }, [toasts])
+
+  useKeyPress(["Escape"], removeOldest)
+
+  useEffect(() => {
+    timer.current = setTimeout(removeOldest, duration)
+    return () => {
+      clearTimeout(timer.current)
+    }
+  }, [toasts, duration, removeOldest])
+
+  const addToast: ContextProps["addToast"] = useCallback(
+    (toast) => {
+      const newToast: ToastType = { ...toast, id: v4() }
+      if (behavior === "stack") {
+        if (position.vertical === "center" || position.vertical === "top") {
+          setToasts((toasts) => [newToast, ...toasts])
+        } else {
+          setToasts((toasts) => [...toasts, newToast])
+        }
```

Comment:
Yeah, you're probably right, since maintaining ordering is really only a visual perk, and isn't necessary from a functionality standpoint. If we programmatically remove a toast, via it's handleclose callback, it doesn't matter much if we maintain the proper order, since it's only ever removing a singular toast by its given id.

Reply:
So actually, I think that we need to maintain `flex-direction: column` to ensure that items appear to enter from the appropriate direction, otherwise they don't look like they're "stacking". I think the fix might be something like this however.

```
const isTopOrCenter =
    position.vertical === "center" || position.vertical === "top"
    
const removeOldest = useCallback(() => {
    if (toasts.length === 0) {
      return
    }
    if (isTopOrCenter) {
      setToasts((toasts) => toasts.slice(0, toasts.length - 1))
    } else {
      setToasts((toasts) => toasts.slice(1))
    }
  }, [toasts, isTopOrCenter])
  
const addToast: ContextProps["addToast"] = useCallback(
  (toast) => {
    const newToast: ToastType = { ...toast, id: v4() }
      if (behavior === "stack") {
        if (isTopOrCenter) {
          setToasts((toasts) => [newToast, ...toasts])
        } else {
          setToasts((toasts) => [...toasts, newToast])
        }
      } else {
        setToasts([newToast])
      }
    },
    [behavior, isTopOrCenter],
  )
```

Reply:
But maybe this is smelly and there is still a more elegant solution.

Reply:
@thispastwinter yeah, this feels pretty complex. So `flex-direction: column` and `flex-direction: column-reverse` don't work identically aside from the order? I haven't played with it, but that's surprising to me. 🤔

Reply:
Let me think through this further and see what I can do 😄

Reply:
No, disregard. I had my logic mixed up earlier when I attempted to just use flex.